docs/viplab3.0/computation_result.md

```diff
@@ -1,141 +1,235 @@
 # Computation Result JSON Message
 
-A *computation result*  is the result of a *computation*.
-It contains information about the success or failure (showing details for students)
-and result objects, like stdout, stderr, images, links, etc.
-
-**This documentation is in progress and contains NOT the final specification**
+A *computation result*  is the result of a *computation*. It always contains
+the stdout and stderr of of the computation. In addition, it may containIt contains 
+arbitrary information contained in one or more artifacts. Artifacts
+are information about the success or failure (showing details for students)
+and result objects, like files, images, links, etc.
 
 ## Example (informal)
 
 ```
-{ "computation" : "4598393-95bf-409a-98a5-ee375982c3e",  // uuid of corresponding computation
+{
+  "identifier" : "86165eea-14df-4a76-805a-09b21441cbf7",
+  "version" : "3.0.0"
+  "computation" : "4598393-95bf-409a-98a5-ee375982c3e",  // uuid of corresponding computation
   "status" : "final", // final and intermediate, 
```

Comment:
Do we need this?

Reply:
Yes. All long living objects have an identifier and version. And from my understanding, computation is needed to be able to route the result back to client who has created the computation.